events/api.py

```diff
@@ -2522,9 +2522,16 @@ def _filter_event_queryset(queryset, params, srs=None):  # noqa: C901
     # Filter by string (case insensitive). This searches from all fields
     # which are marked translatable in translation.py
 
+    # filter to get events with a registration.
     val = params.get("registration", None)
     if val:
         queryset = queryset.exclude(registration=None)
+
+    # filter to get events without a registration.
+    val = params.get("registration!", None)
+    if val:
+        queryset = queryset.filter(registration=None)
```

Comment:
It seems that using an exclamation mark is an existing pattern here, so I'm not going to say anything about that. Though I find it a bit misleading that you can get the same results/filtering with `registration=true` and `registration=false`,

Maybe the whole thing could be replaced with something like this?


```suggestion
    # filter to get events with or without a registration.
    val = params.get("registration", None)
    if val and parse_bool(val):
        queryset = queryset.exclude(registration=None)
    elif val: 
        queryset = queryset.filter(registration=None)
```

Reply:
Very good idea. Changed as suggested, fixed the tests and updated the documentation

Reply:
Just a general note, this "do something if value is defined regardless of what the value is" elsewhere also an existing pattern linkedevents. But  :+1: for avoiding that.